File: Modules/C#/Day15/Milit-Panchasara/Assignment/Controllers/EmployeesController.cs
using Assignment1.Models;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Assignment1.Controllers
{
    [Route("emps")]
    [ApiController]
    public class EmployeesController : ControllerBase
    {
        private readonly CSDay15Context context;

        public EmployeesController(CSDay15Context context)
        {
            this.context = context;
        }

        [HttpGet]
        public ActionResult<IEnumerable<Employee>> Get()
        {
            return context.Employees.ToList();
        }

        [HttpGet("{id}")]
        public ActionResult<Employee?> Show(int id)
        {
            return context.Employees.SingleOrDefault(x => x.Id == id);
        }

        [HttpPost]
        public ActionResult<Employee?> Add(Employee request)
        {
            var emp = new Employee();
            emp.AddressLine1 = request.AddressLine1;
            emp.AddressLine2 = request.AddressLine2;
            emp.AddressLine3 = request.AddressLine3;
            emp.Assignments = request.Assignments;
            emp.CitizenshipId = request.CitizenshipId;
            emp.CitizenshipLegislationCode = request.CitizenshipLegislationCode;
            emp.CitizenshipStatus = request.CitizenshipStatus;
            emp.CitizenshipToDate = request.CitizenshipToDate;
            emp.City = request.City;
            emp.CorrespondenceLanguage = request.CorrespondenceLanguage;
            emp.Country = request.Country;
            emp.CreationDate = request.CreationDate;
            emp.DateOfBirth = request.DateOfBirth;
            emp.DisplayName = request.DisplayName;
            emp.EffectiveStartDate = request.EffectiveStartDate;
            emp.FirstName = request.FirstName;
            emp.Gender = request.Gender;
            emp.HomePhoneAreaCode = request.HomePhoneAreaCode;
            emp.HireDate = request.HireDate;
            emp.HomePhoneCountryCode = request.HomePhoneCountryCode;
            emp.HomePhoneExtension = request.HomePhoneExtension;
            emp.HomePhoneLegislationCode = request.HomePhoneLegislationCode;
            emp.HomePhoneNumber = request.HomePhoneNumber;
            emp.Honors = request.Honors;
            emp.LastName = request.LastName;
            emp.LastUpdateDate = null;
            emp.LegalEntityId = request.LegalEntityId;
            emp.LicenseNumber = request.LicenseNumber;
            emp.MaritalStatus = request.MaritalStatus;
            emp.MiddleName = request.MiddleName;
            emp.NameSuffix = request.NameSuffix;
            emp.NationalId = request.NationalId;
            emp.NationalIdCountry = request.NationalIdCountry;

            context.Employees.Add(emp);
            context.SaveChanges();

            return emp;
        }

        [HttpPut("{id}")]
        public ActionResult<Employee?> Update(int id, Employee request)
        {
            var emp = context.Employees.SingleOrDefault(x => x.Id == id);
            if (emp == null)
            {
                return null;
            }
            emp.AddressLine1 = request.AddressLine1;
            emp.AddressLine2 = request.AddressLine2;
            emp.AddressLine3 = request.AddressLine3;
            emp.Assignments = request.Assignments;
            emp.CitizenshipId = request.CitizenshipId;
            emp.CitizenshipLegislationCode = request.CitizenshipLegislationCode;
            emp.CitizenshipStatus = request.CitizenshipStatus;
            emp.CitizenshipToDate = request.CitizenshipToDate;
            emp.City = request.City;
            emp.CorrespondenceLanguage = request.CorrespondenceLanguage;
            emp.Country = request.Country;
            emp.CreationDate = request.CreationDate;
            emp.DateOfBirth = request.DateOfBirth;
            emp.DisplayName = request.DisplayName;
            emp.EffectiveStartDate = request.EffectiveStartDate;
            emp.FirstName = request.FirstName;
            emp.Gender = request.Gender;
            emp.HomePhoneAreaCode = request.HomePhoneAreaCode;
            emp.HireDate = request.HireDate;
            emp.HomePhoneCountryCode = request.HomePhoneCountryCode;
            emp.HomePhoneExtension = request.HomePhoneExtension;
            emp.HomePhoneLegislationCode = request.HomePhoneLegislationCode;
            emp.HomePhoneNumber = request.HomePhoneNumber;
            emp.Honors = request.Honors;
            emp.LastName = request.LastName;
            emp.LastUpdateDate = DateTime.Now;
            emp.LegalEntityId = request.LegalEntityId;
            emp.LicenseNumber = request.LicenseNumber;
            emp.MaritalStatus = request.MaritalStatus;
            emp.MiddleName = request.MiddleName;
            emp.NameSuffix = request.NameSuffix;
            emp.NationalId = request.NationalId;
            emp.NationalIdCountry = request.NationalIdCountry;

            context.SaveChanges();

            return emp;
        }
    }
}

File: Modules/NodeJS/Day11/Shekhar chawla/Assignment/controllers/employees.controller.js
const Employees = require('../models/employees/employees.models')

class EmpController {
    getAll(req, res) {
        Employees.find((err, employees) => {
            if (err) return res.send(err)
            return res.json(employees)
        })
    }

    get(req, res) {
        Employees.findById(req.params.id, (err, employee) => {
            if (err) return res.status(500).send(err)
            return res.json(employee)
        })
    }

    create(req, res) {
        const employee = new Employees(req.body)

        employee.save((err, result) => {
            if (err) {
                console.log(err)
                return res.status(500).send(err)
            }

            return res.json({ msg: "Employee created", employee: result })
        })

    }


    update(req, res) {
        Employees.findByIdAndUpdate(req.params.id, req.body, { new: true }, (err, e) => {
            if (err) {
                console.log(err)
                return res.send('error')
            }
    
            return res.json({ msg: 'Employee Updated', Employee: e })
        })

    }


    delete(req, res) {
        Employees.findByIdAndRemove(req.params.id, (err, e) => {
            if (err) {
                console.log(err)
                return res.send('error')
            }
    
            return res.json({ msg: 'Employees Deleted', Employee: e })
        })
    }

    
}


module.exports = EmpController;




/*
var e1 = new Employees(
    {
        AddressLine1: "D-45 Gandhi houses",
        AddressLine2: "near kasturba flats",
        AddressLine3: "sabarmati ashram",
        CitizenshipId: 1,
        CitizenshipLegislationCode: "A1",
        CitizenshipStatus: "Indian",
        CitizenshipToDate: '2002-12-09',
        City: "Ahmedabad",
        CorrespondenceLanguage: "Gujarati",
        Country: "India",
        CreationDate: '2002-12-09',
        DateOfBirth: '2002-12-09',
        DisplayName: "bapu",
        DriversLicenseExpirationDate: '2002-12-09',
        DriversLicenseId: 123456,
        DriversLicenseIssuingCountry: "India",
        EffectiveStartDate: '2002-12-09',
        Ethnicity: "Indo-Aryan",
        FirstName: "Mohandas",
        Gender: "Male",
        HireDate: '2002-12-09',
        HomeFaxAreaCode: "A4552",
        HomeFaxCountryCode: "I15545",
        HomeFaxExtension: ".pdf",
        HomeFaxLegislationCode: "D554",
        HomeFaxNumber: "1225455",
        HomePhoneAreaCode: "A455522",
        HomePhoneCountryCode: "012",
        HomePhoneExtension: "48485",
        HomePhoneLegislationCode: "4542200",
        HomePhoneNumber: "7990165272",
        Honors: "Mr",
        LastName: "Gandhi",
        LastUpdateDate: '2002-12-09',
        LegalEntityId: 54811,
        LicenseNumber: "FFEE0215454",
        MaritalStatus: "Married",
        MiddleName: "Karamchand",
        MilitaryVetStatus: "No",
        NameSuffix: "bapu",
        NationalId: "264644",
        NationalIdCountry: "99",
        assignments: [
            {
                ActionCode: "1",
                ActionReasonCode: "1",
                ActualTerminationDate: '2002-12-09',
                AssignmentCategory: "Freedom",
                AssignmentId: "1",
                AssignmentName: "Salt Movement",
                AssignmentNumber: "111",
                AssignmentProjectedEndDate: '2002-12-09',
                AssignmentStatus: "Completed",
                AssignmentStatusTypeId: 11223344,
                BusinessUnitId: 11223344,
                CreationDate: '2002-12-09',
                DefaultExpenseAccount: "Yes",
                DepartmentId: 11223344,
                EffectiveEndDate: '2002-12-09',
                EffectiveStartDate: '2002-12-09',
                EndTime: "1 :4 :12",
                Frequency: "15",
                FullPartTime: "Yes",
                GradeId: 11223344,
                GradeLadderId: 11223344,
                JobId: 11223344,
                LastUpdateDate: '2002-12-09',
                LegalEntityId: 11223344,
                LocationId: 11223344,
                ManagerAssignmentId: 11223344,
                ManagerId: 11223344,
                assignmentDFF: [],
                assignmentExtraInformation: [],
                empreps: [],
                links: [
                    "Sardar Patel"
                ]
            },
            {
                ActionCode: "2",
                ActionReasonCode: "2",
                ActualTerminationDate: '2002-12-09',
                AssignmentCategory: "Freedom",
                AssignmentId: "2",
                AssignmentName: "Dandi March",
                AssignmentNumber: "222",
                AssignmentProjectedEndDate: '2002-12-09',
                AssignmentStatus: "Completed",
                AssignmentStatusTypeId: 55667788,
                BusinessUnitId: 55667788,
                CreationDate: '2002-12-09',
                DefaultExpenseAccount: "Yes",
                DepartmentId: 55667788,
                EffectiveEndDate: '2002-12-09',
                EffectiveStartDate: '2002-12-09',
                EndTime: "1 :4 :12",
                Frequency: "15",
                FullPartTime: "Yes",
                GradeId: 55667788,
                GradeLadderId: 55667788,
                JobId: 55667788,
                LastUpdateDate: '2002-12-09',
                LegalEntityId: 55667788,
                LocationId: 55667788,
                ManagerAssignmentId: 55667788,
                ManagerId: 55667788,
                assignmentDFF: [],
                assignmentExtraInformation: [],
                empreps: [],
                links: [
                    "Sardar Patel"
                ]
            }
        ],
        directReports: [
            "school certificate",
            "driving license"
        ],
        links: [
            "bhagat singh",
            "sukhdev",
            "ambedkar doctor"
        ]
    }
)
*/
// e1.save()
//     .then( result => console.log(result))
//     .catch( err => console.log(err))
